pkg/config/db.go

```diff
@@ -0,0 +1,134 @@
+package config
+
+import (
+	v1API "github.com/supabase/cli/pkg/api"
+	"github.com/supabase/cli/pkg/cast"
+	"github.com/supabase/cli/pkg/diff"
+)
+
+type (
+	PoolMode string
+
+	// All of thoses are remote only settings that'll apply to supabase hosted database
+	remoteDb struct {
```

Comment:
**remark**

I've also considered moving this under `[experimental.db]` config instead, maybe that will be more suitable ? Since we'll need to update the docs for it anyways 🤔

Reply:
We can probably nest it under `[db.settings]`. It would nicely solve https://github.com/supabase/cli/issues/2611 once we support it locally.

Reply:
Yup remembered I saw something like this somewhere. Gonna put that in my todo list thank's for pointing it out.